### PR TITLE
adds temporal cli compat

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Review by shivanshuraj1333 — APPROVED** [review]
LGTM

**Review by shivanshuraj1333 — CHANGES_REQUESTED** [review]
CI job for CVE is failing

**Comment by mritunjaysharma394** [comment]
Yes @shivanshu1333, that's why the adv PR and this comment: https://github.com/wolfi-dev/os/pull/9726#issuecomment-1850117358

File: temporal.yaml [hunk 1]
```diff
@@ -42,6 +42,14 @@ subpackages:
           ldflags: -s -w
       - uses: strip
 
+  - name: temporal-compat
+    description: "Compat package for temporal"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/local/bin
+          ln -s /usr/bin/temporal "${{targets.contextdir}}"/usr/local/bin
+      - uses: strip
```
Comment (stormqueen1990) [inline temporal.yaml:51]: This is just a symlink. I don't think we need `uses: strip`.

Comment (mritunjaysharma394) [reply temporal.yaml:51]: Oh yes, it was a copy pasta and it's happening at many places so I think we need to progressively fix that too, let me fix it here for now, thanks!

File: temporal.yaml
[the remark this answers is not in the document]
Comment (mritunjaysharma394) [reply temporal.yaml:1]: Not that I am aware of, I think this package already existed a lot earlier in its current state and don't know why it was done or added then, I can fix this too

File: temporal.yaml (hunk 1 above)
Comment (mritunjaysharma394) [reply temporal.yaml]: fixed, thanks

File: temporal.yaml
[the remark this answers is not in the document]
Comment (mritunjaysharma394) [reply temporal.yaml:1]: Fixed

**Review by stormqueen1990 — APPROVED** [review]

**Review by shivanshuraj1333 — APPROVED** [review]

**Pull request merged**